packages/hardhat-core/src/builtin-tasks/node.ts

```diff
@@ -66,6 +51,122 @@ Private Key: ${privateKey}
   }
 }
 
+subtask(TASK_NODE_GET_PROVIDER)
+  .addParam("forkUrl", undefined, undefined, types.string)
+  .addParam("forkBlockNumber", undefined, undefined, types.int)
+  .setAction(
+    async (
+      {
+        forkBlockNumber: forkBlockNumberParam,
+        forkUrl: forkUrlParam,
+      }: {
+        forkBlockNumber?: number;
+        forkUrl?: string;
+      },
+      { artifacts, config, network }
+    ): Promise<EthereumProvider> => {
+      if (network.name !== HARDHAT_NETWORK_NAME) {
+        const networkConfig = config.networks[HARDHAT_NETWORK_NAME];
+
+        return lazyObject(() => {
```

Comment:
I don't think we need a `lazyObject` here, as it will be immediately used.

Reply:
Also, the `return` prevents the fork options from working.

Reply:
Is it though? In this scenario, the provider is returned and used exactly as before (the server constructor receives it, and this in turn uses it to create the handler, that doesn't immediately uses the provider).

I'm ok with not making it lazy anyway, since I don't think for the node task this matter, but just wanted to point out that the behavior is equivalent to the current one.